lci_strategic_plugin/include/lci_strategic_plugin/lci_strategic_plugin.h

```diff
@@ -125,7 +135,78 @@ class LCIStrategicPlugin
    * \brief Lookup transfrom from front bumper to base link
    */
   void lookupFrontBumperTransform();
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+  /**
+   * \brief callback function for mobility operation
+   * 
+   * \param msg input mobility operation msg
+   */
+  void mobilityOperationCb(const cav_msgs::MobilityOperationConstPtr& msg);
+
+  /**
+   * \brief BSM callback function
+   */
+  void BSMCb(const cav_msgs::BSMConstPtr& msg);
 
+ /**
+   * \brief parse strategy parameters
+   * 
+   * \param strategy_params input string of strategy params from mob op msg
+   */
+  void parseStrategyParams(const std::string& strategy_params);
+
+  /**
+   * \brief Generates Mobility Operation messages
+   *
+   * \return mobility operation msg for status and intent
+   */
+  cav_msgs::MobilityOperation generateMobilityOperation();
+
+  /**
+   * \brief Determine the turn direction at intersection
+   *
+   * \param lanelets_list List of lanelets crossed around the intersection area
+   *
+   * \return turn direction in format of straight, left, right
+   *
+   */
+  TurnDirection getTurnDirectionAtIntersection(std::vector<lanelet::ConstLanelet> lanelets_list);
+
+  /**
+   * \brief Method to call at fixed rate in execution loop. Will publish plugin discovery and mobility operation msgs.
+   * 
+   * \return True if the node should continue running. False otherwise
+   */ 
+  bool onSpin();
+
+
+  ros::Publisher mobility_operation_pub;
+  
+  ////////// VARIABLES ///////////
+
+  TurnDirection intersection_turn_direction_ = TurnDirection::Straight;
+  bool approaching_light_controlled_interction_ = false;
+
+  // CARMA Streets Variakes
+  // timestamp for msg received from carma streets
+  unsigned long long street_msg_timestamp_ = 0;
+  // scheduled stop time
+  unsigned long long scheduled_stop_time_ = 0;
+  // scheduled enter time
+  unsigned long long scheduled_enter_time_ = 0;
+  // scheduled depart time
+  unsigned long long scheduled_depart_time_ = 0;
+  // scheduled latest depart position
+  uint32_t scheduled_departure_position_ = std::numeric_limits<uint32_t>::max();
+  // flag to show if the vehicle is allowed in intersection
+  bool is_allowed_int_ = false;
+  
+  //BSM
+  std::string bsm_id_ = "default_bsm_id";
+  uint8_t bsm_msg_count_ = 0;
+  uint16_t bsm_sec_mark_ = 0;
```

Comment:
Any particular reason for making these variables public when the others are private?